Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -361,31 +448,208 @@ ReactantProductAtomMapping *getAtomMappingsReactantProduct(
   return mapping;
 }
 
+namespace {
+unsigned reactProdMapAnchorIdx(Atom *atom, const RDKit::UINT_VECT &pMatches) {
+  PRECONDITION(atom, "no atom");
+  if (pMatches.size() == 1) {
+    return pMatches[0];
+  }
+
+  const auto &pMol = atom->getOwningMol();
+  const unsigned atomIdx = atom->getIdx();
+
+  auto areAtomsBonded = [&pMol, &atomIdx](const unsigned &pAnchor) {
+    return pMol.getBondBetweenAtoms(atomIdx, pAnchor) != nullptr;
+  };
+
+  auto match = std::find_if(pMatches.begin(), pMatches.end(), areAtomsBonded);
+
+  CHECK_INVARIANT(match != pMatches.end(), "match not found");
+  return *match;
+}
+
+void forwardReactantBondStereo(ReactantProductAtomMapping *mapping, Bond *pBond,
+                               const ROMol &reactant, const Bond *rBond) {
+  PRECONDITION(mapping, "no mapping");
+  PRECONDITION(pBond, "no bond");
+  PRECONDITION(rBond, "no bond");
+  PRECONDITION(rBond->getStereo() > Bond::BondStereo::STEREOANY,
+               "bond in reactant must have defined stereo");
+
+  auto &prod2React = mapping->prodReactAtomMap;
+  auto &react2Prod = mapping->reactProdAtomMap;
+
+  const Atom *rStart = rBond->getBeginAtom();
+  const Atom *rEnd = rBond->getEndAtom();
+  const auto &rStereoAtoms = rBond->getStereoAtoms();
+
+  StereoBondEndCap start(reactant, rStart, rEnd, rStereoAtoms[0]);
+  StereoBondEndCap end(reactant, rEnd, rStart, rStereoAtoms[1]);
+
+  // The bond might be matched backwards in the reaction
+  if (prod2React[pBond->getBeginAtom()->getIdx()] == rEnd->getIdx()) {
+    std::swap(start, end);
+  } else if (prod2React[pBond->getBeginAtom()->getIdx()] != rStart->getIdx()) {
+    throw std::logic_error("Reactant and Product bond ends do not match");
+  }
+
+  auto rStartAnchorMatches = react2Prod.find(start.getAnchorIdx());
+  auto rEndAnchorMatches = react2Prod.find(end.getAnchorIdx());
+
+  // If any of the anchor pair is not mapped, see if we can make the match with
```

Comment:
It might help to have an example reaction smarts of these in the comments (and below)l

Reply:
I changed this to also consider "stereoatom, non-stereoatom" pair stereo forwarding, and added a long comment with examples.